pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -1134,7 +1143,6 @@ protected void replaceDocId(IndexSegment newSegment, ThreadSafeMutableRoaringBit
         doBatchRefreshUpsertView(_upsertViewRefreshIntervalMs);
       }
       if (_enableSnapshot) {
-        _updatedSegmentsSinceLastSnapshot.add(newSegment);
```

Comment:
no need to add `new` segment to Set as we have tracked it in the higher level methods, but we should track the old segment being updated while adding docs from the new segment. 

similar for the removal of following two methods that update `new` segment, which is tracked in Set already.